FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift

```diff
@@ -12,162 +12,179 @@
-      imageView.widthAnchor.constraint(equalToConstant: 325),
-      imageView.heightAnchor.constraint(equalToConstant: 325),
-    ])
-  }
-
-  private func setupEmailTextfield() {
-    emailTextField = textField(placeholder: "Email", symbolName: "person.crop.circle")
-    emailTextField.translatesAutoresizingMaskIntoConstraints = false
-    addSubview(emailTextField)
-    NSLayoutConstraint.activate([
-      emailTextField.leadingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.leadingAnchor,
-        constant: 15
-      ),
-      emailTextField.trailingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.trailingAnchor,
-        constant: -15
-      ),
-      emailTextField.heightAnchor.constraint(equalToConstant: 45),
-    ])
-
-    let constant: CGFloat = UIDevice.current.orientation.isLandscape ? 15 : 50
-    emailTopConstraint = emailTextField.topAnchor.constraint(
-      equalTo: safeAreaLayoutGuide.topAnchor,
-      constant: constant
-    )
-    emailTopConstraint.isActive = true
-  }
-
-  private func setupPasswordTextField() {
-    passwordTextField = textField(placeholder: "Password", symbolName: "lock.fill")
-    passwordTextField.translatesAutoresizingMaskIntoConstraints = false
-    addSubview(passwordTextField)
-    NSLayoutConstraint.activate([
-      passwordTextField.leadingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.leadingAnchor,
-        constant: 15
-      ),
-      passwordTextField.trailingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.trailingAnchor,
-        constant: -15
-      ),
-      passwordTextField.heightAnchor.constraint(equalToConstant: 45),
-    ])
-
-    let constant: CGFloat = UIDevice.current.orientation.isLandscape ? 5 : 20
-    passwordTopConstraint =
-      passwordTextField.topAnchor.constraint(
-        equalTo: emailTextField.bottomAnchor,
-        constant: constant
-      )
-    passwordTopConstraint.isActive = true
-  }
-
-  private func setupLoginButton() {
-    addSubview(loginButton)
-    loginButton.translatesAutoresizingMaskIntoConstraints = false
-    NSLayoutConstraint.activate([
-      loginButton.leadingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.leadingAnchor,
-        constant: 15
-      ),
-      loginButton.trailingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.trailingAnchor,
-        constant: -15
-      ),
-      loginButton.heightAnchor.constraint(equalToConstant: 45),
-      loginButton.centerYAnchor.constraint(equalTo: centerYAnchor, constant: 5),
-    ])
-  }
-
-  private func setupCreateAccountButton() {
-    addSubview(createAccountButton)
-    createAccountButton.translatesAutoresizingMaskIntoConstraints = false
-    NSLayoutConstraint.activate([
-      createAccountButton.centerXAnchor.constraint(equalTo: centerXAnchor),
-      createAccountButton.topAnchor.constraint(equalTo: loginButton.bottomAnchor, constant: 5),
-    ])
+// TODO(ncooke3): Use view modifiers?
```

Comment:
Or implement a custom button style.

Reply:
Yes, a button style sounds good. When I tried to create a very simple button style, the custom style lost the button's default ability to change text color on tap. I feel like I'm not doing something right.

```swift
private struct MyButtonStyle : ButtonStyle {
  func makeBody(configuration: Configuration) -> some View {
    configuration.label
  }
}

#Preview {
  Button("Hello") {}
  .buttonStyle(MyButtonStyle())
}
```

<img width="373" alt="Screenshot 2024-10-14 at 12 59 42 PM" src="https://github.com/user-attachments/assets/807fe4f6-7e0f-44c5-b6ae-89c709c96ef3">

Should I leave this as is? What's the most idiomatic pattern?